daemon/cluster/cluster.go

```diff
@@ -622,6 +623,12 @@ func (c *Cluster) CreateService(s types.ServiceSpec) (string, error) {
 	if err != nil {
 		return "", err
 	}
+
+	if auth, ok := headers["x-registry-auth"]; ok {
+		fmt.Println("AUTH", auth[0])
+		// TODO(nishanttotla): Add auth[0] to the ContainerSpec before passing to swarm
+	}
```

Comment:
@tonistiigi need some help here. Is the `ContainerSpec` expected to exist here, so that I can put the auth token inside it? See https://github.com/docker/swarmkit/pull/965 for reference.